Server/GameServer/User.cpp

```diff
@@ -1180,6 +1180,37 @@ void CUser::SendTime()
 }
 
 /**
+* @brief Overloads CUser::SendTime() to change the server time by GM command.
+*/
+void CUser::SendTime(Packet & pkt)
+{
+	if (!this->isGM()) return;
+
+	int16_t year  = pkt.read<int16_t>();
+	int16_t month = pkt.read<int16_t>();
+	int16_t day   = pkt.read<int16_t>();
+	int16_t hour  = pkt.read<int16_t>();
+	int16_t min   = pkt.read<int16_t>();
+
+	if (year  != NULL) g_pMain->m_sYear  = year;
+	if (month != NULL) g_pMain->m_sMonth = month;
+	if (day   != NULL) g_pMain->m_sDate  = day;
+	if (hour  != NULL) g_pMain->m_sHour  = hour;
+	if (min   != NULL) g_pMain->m_sMin   = min;
+
+	if (year == NULL && month == NULL && day == NULL && hour == NULL && min == NULL)
+	{
+		g_pMain->SendHelpDescription(this, "Using Sample: /time 18:00 2002.7.1 (Using the second parameter is not must(date format: Year.Month.Day).");
+		return;
+	}
+
+	Packet result(WIZ_TIME);
+	result << uint16_t(g_pMain->m_sYear) << uint16_t(g_pMain->m_sMonth) << uint16_t(g_pMain->m_sDate)
+		   << uint16_t(g_pMain->m_sHour) << uint16_t(g_pMain->m_sMin);
+	Send(&result);
```

Comment:
I think we could probably just call SendTime() here instead of redoing this packet, but that's just a minor nitpick. :)

Reply:
The reason for overloading this function is because CUser::GameStart(Packet & pkt) already using it.
I though it already updates the server time for all players? as g_pMain is a pointer to CGameServerDlg(if that's what you were referring to CEbenezerDlg?).

Reply:
No, it updates the server time but it only immediately notifies the player who changed the time. You need to send this packet to all in-game players for the change to be made to those players.

SendTime() merely sends the packet containing the time info to the clients. This is why I said that instead of reimplementing this logic to send this (same) packet to the player, it would've been preferable to just have your method call the existing method to do so (no point duplicating logic, after all).

And yes, "CGameServerDlg" is CEbenezerDlg. Somewhere along the line someone renamed this.

Reply:
Oh...I think i get what you're saying. So the /time command doesn't really updates the server time? It's just sends a packet to the server and the server sends back to the client so the weather basically changes.
I'll try to look at it now and update the pull request.

Reply:
I just realized that the CGameServerDlg::UpdateGameTime get executed on a background thread that gets the local time of the server machine every 6 seconds from DateTime wrapper class.
Which is why it got me confused and i implemented the /time command to parse also date(which is absolutely not necessary in this case).
But i commented it out in case someone will want in the future to parse date also for whatever reason and the server will support it if needed.
Will update the pull request soon.